hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/inventory/Operation.java

```diff
@@ -16,18 +16,12 @@
  */
 package org.hawkular.agent.monitor.inventory;
 
-public abstract class Operation<RT extends ResourceType<?, ?, ?, ?>> extends NamedObject {
+public abstract class Operation extends NamedObject {
 
-    private final RT resourceType;
     private String operationName;
 
-    public Operation(ID id, Name name, RT resourceType) {
```

Comment:
I removed it mainly, because I have not seen that it would be used now. 

The second, general point is that `resourceType` inside `Operation` brings a circular reference chain between `ResourceType` and `Operation` which is mostly not a good idea. E.g. it is not possible to make both classes immutable.

Thirdly, I thought that we probably won't need the link to `ResourceType`  in `Operation` if we can do well without it in `MetricType` and `AvialType`.

The same holds for `ResourceConfigurationPropertyType`.

Anyway, if there comes a situation, where the link will be inevitable, I am ready to introduce it back. But I still vote for removing it now.